### PR TITLE
Kdtree cleanup

### PR DESCRIPTION
**Pull request by mdehoon**
This pull request is a follow-up of pull request #1580.

This pull request moves Bio.KDTree to a new module Bio.PDB.kdtrees, and declares the old module Bio.KDTree obsolete.
Functionality is the same, but the code has been cleaned up.
In particular, the old code had four things called KDTree:
- `Bio/KDTree/__init__.py`
- `Bio/KDTree/KDTree.py`
- The class `KDTree` inside `Bio/KDTree/KDTree.py`
- The class `KDTree` inside `Bio/KDTree/KDTreemodule.c`

The new code consists of a single module `kdtrees`, and implements a single class `KDTree`.

As KDTree objects are only used in `Bio/PDB/NeighborSearch.py`, the new module `kdtrees` was placed in `Bio/PDB/`, rather than a top-level module of `Bio`.

The new module is tested in `Tests/test_PDB_KDTree.py`, which was derived from (but is more extensive than) `Tests/test_KDTree.py`. The testing code in `Tests/test_KDTree.py` can remain until `Bio.KDTree` is removed from Biopython.

I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and
TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style
checks pass with these changes.

I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/PDB/kdtrees.c [hunk 1]
```diff
@@ -0,0 +1,1483 @@
+#include <math.h>
```
Comment (peterjc) [inline Bio/PDB/kdtrees.c]: Can you put an explicit dual license comment header in here? Not sure if we've dual licensed any of the C or C++ code yet, so you may get to set the convention for this.

**Comment by peterjc** [comment]
Could you also declare ``Bio.KDTree`` obsolete in the ``DEPRECATED.rst`` file, directing people at the new module instead?

**Comment by peterjc** [comment]
I don't think we can easily measure the C or C++ code coverage line by line, but the Python side test coverage seems fine.

**Comment by peterjc** [comment]
Assuming there are no objections to putting this under ``Bio.PDB``, this seems fine. I have not looked at the implementation details (Python and C not being something I'm an expert in).

**Comment by mdehoon** [comment]
@JoaoRodrigues  Any objections against putting the kdtrees module in Bio.PDB? Since kdtrees is used only in Bio.PDB, it seems the right place for it.

**Comment by peterjc** [comment]
(I was thinking people might be using the KD Trees code with other data outside of PDB structures)

**Comment by mdehoon** [comment]
@peterjc  That is possible in principle, but probably unlikely. There is no documentation for KDTree in the Biopython tutorial, and I haven't seen any bug reports where people are using KDTree outside of Bio.PDB.

**Comment by peterjc** [comment]
Thank you!

**Pull request merged**